lib/metasploit/framework/login_scanner/snmp.rb

```diff
@@ -347,10 +388,20 @@ def parse_snmp_response(pkt)
         # Create a new socket for this scanner
         def configure_socket
           shutdown_socket if sock
-          self.sock = ::Rex::Socket::Udp.create(
+
+          klass = ::Rex::Socket::Udp
+
+          if protocol == 'tcp'
+            klass = ::Rex::Socket::Tcp
+          end
+
+          self.sock = klass.create({
```

Comment:
Rex::Socket actually provides a `Proto` [argument](https://github.com/rapid7/rex-socket/blob/60103dbbd5f540f8d11d5b69417f2ad9d7ae2d1b/lib/rex/socket/parameters.rb#L70) for exactly this purpose. You should be able to just call `::Rex::Socket.create` and pass `Proto` in the hash as either tcp or udp.

Reply:
Cool, didn't know that - documentation for Rex::Socket is a bit scarce - thank you for pointing that out